File: spC++/src/Implement/Models/TableOfContents.cpp
#include "Headers/Models/TableOfContents.hpp"

TableOfContents::TableOfContents(TableOfContents* toc) {
    this->entries = toc->entries;
}

void TableOfContents::accept(IVisitor* v) {
    v->visitToC(this);
}

void TableOfContents::addEntry(std::string entry) {
    this->entries.push_back(entry);
}

IElement *TableOfContents::clone() {
    return new TableOfContents(this);
}

std::vector<std::string> TableOfContents::getEntries() {
    return this->entries;
}

File: Java-Labs/lab/src/main/java/sp/lab/Models/TableOfContents.java
package sp.lab.Models;

import java.util.ArrayList;
import java.util.List;

public class TableOfContents extends Element{
    private List<String> entries;

    public TableOfContents() {
        this.entries = new ArrayList<>();
    }

    public void addEntry(String entry) {
        this.entries.add(entry);
    }

    @Override
    public void accept(IVisitor visitor) {
        visitor.visitTableOfContents(this);
    }

    @Override
    public void print() {
        System.out.println("Table of contents: ");
        for (String entry : this.entries) {
            System.out.println(entry);
        }
    }
}
